pure3.md

```diff
@@ -0,0 +1,182 @@
+level Targets role to issue new metadata for every update, Uptane's delegations
+model can be used to make this entire process more scalable. Instead, delegated
+roles can be used to sign for the image itself and its in-toto link metadata.
+However, in-toto layouts and the key artifacts used to verify them should be
+signed for by the top level Targets role.
+
+Once the Director repository decides which images are required and signs Targets
+metadata for them, the vehicle must fetch the images and all the metadata using
+the interfaces provided by both the repositories. In this phase, the Director
+repository can continue signing targets metadata for the images, as per the
+default functioning of Uptane, and serve the metadata to the automobile. The
+automobile fetches the image itself from the Image repository along with its
+corresponding in-toto metadata.
+
+TODO: talk about role / delegations setups with a diagram / table
+
+### Full Verification ECUs
+
+The Primary ECU receives images and metadata from the repositories, and sends
+them on to the other ECUs that it updates. Each Full Verification ECU is
+extended to perform independent in-toto verification that will be secure even
+if the Primary ECU is compromised. Some Full Verification ECUs, for example the
+Primary, are also expected to perform in-toto verification on behalf of Partial
+Verification ECUs that are resource constrained.
+
+For every target image the client needs to install, the corresponding in-toto
+layout, links, and keys are identified via the custom field of the existing
+Uptane metadata. These files are available as targets on the Image repository.
+Therefore, for each installable target, the updater will need to download and
+verify the Uptane metadata that signs for the in-toto metadata, either the
+Targets role or a delegated role. If verification passes, the actual in-toto
+metadata files—layout, links, and public keys—must be downloaded and their
+hashes must be compared with the Uptane metadata. With these two checks,
+Uptane’s secure delivery properties are extended to the in-toto files. These
+files can then be used with confidence by the client to perform in-toto
+verification.
+### Resource Constrained ECUs
+
+The addition of in-toto metadata generates additional overhead, which can be
+problematic for resource constrained ECUs. Uptane already accounts for resource
+constrained secondary ECUs by providing a separate Partial Verification
+workflow. This document assumes that such ECUs are incapable of performing
+in-toto verification and therefore will expect other capable ECUs, perhaps the
+Primary ECU, to perform in-toto verification on their behalf. In this model,
+while every ECU may not be verifying the software supply chain of the images it
+is expected to install, verification is still happening on the individual
+automobile. Essentially, there is no difference in the functioning of Partial
```

Comment:
This isn't quite true. The partial verification secondaries are relying on the security on the primary

Reply:
Just took another pass at this paragraph, emphasizing the reliance on another ECU for supply chain verification. Also replaced "functioning" with "update workflow" because these constrained ECUs don't actually have to do anything different. Does that help?